### DISCUSSION TITLE
TLS Negotiation failed: FAILED_PRECONDITION: starttls error (0)

### DISCUSSION THREAD

**Discussion by dpushkar24** [postalserver/postal#2244] [Help with using Postal]
Hi Team,

I have gone through multiple blogs from your team and would like to congratulate for your work and efforts :)

Actually, what brings me here is, I am facing an issue while receiving mails from Gmail. However my mail server is able to revceive mails from almost all other domains like rediffmail, (some verification mails from) cloudflare, and my multiple clients too.
following is the screenshot I am getting;
![image](https://user-images.githubusercontent.com/24427158/202978156-af520d01-582c-4a28-838e-67ba79908d64.png)

I am using hestiacp tool for hosting and hosted on Oracle free tier cloud. However I am using cloudflare as my certification authority and all the mail related entries are kept on DNS only;
![image](https://user-images.githubusercontent.com/24427158/202978823-693763cb-f67f-4800-899a-8efded34b83c.png)

I also understand Oracle is blocking my port 25 hence using SMTP relay for sending mails, which is working perfectly fine. 
Also my one strange thing I came across is when I try to ping my mail server (mail.domain.com) it is not reachable at all but mails are coming in (except from Gmail)
![image](https://user-images.githubusercontent.com/24427158/202979478-f36e1e15-7a7b-4474-ab35-340391cfa95d.png)

So what I want my server to just receive mail from Gmail.
So Just wanted to know if I am missing any settings in NGINX? And what will be the directory file path for SMTP settings from backed?
Dose Postal SMTP are supported by Hestiacp? Any separate installation needed if it is really required in this case?

Sorry if you find any silly questions, I am not an system admin.

Any help and support will be really appreciated.

Thanks and Regards,
Pushkar Deshpande

**Comment by willpower232** [comment]
If cloudflare is doing the majority of your certificate work, you will have to generate your own and provide it to Postal directly if you haven't already.

You should be able to verify your SSL status with `openssl s_client -connect mail.yourpostal.com:25 -starttls smtp`

For basic connection tests, `ping` may or may not work depending how the server is set up, the real test would be `telnet mail.yourpostal.com 25` to confirm how open it is.

**Reply by dpushkar24** [reply]
Hi Willpower232,

Thanks for quick response.
I am attaching a response of my SSL Status in the attachment.

However, telnet mail.yourpostal.com 25 will not work in my case as my default port 25 is blocked :(
![11111](https://user-images.githubusercontent.com/24427158/203248059-bd17c74a-f0fe-401f-8bf2-76038961dc81.png)
![2222](https://user-images.githubusercontent.com/24427158/203248070-b35dc8b4-87a0-400f-9cd8-3f0a33edbeca.png)

Looking forward for more inputs.

Once again thank you so much for help.

Thanks and Regards,
dpushkar24

**Reply by willpower232** [reply]
well you just read the certificate from port 25 so maybe not as blocked as you might think? either way, if oracle is blocking the port then I guess that would explain what the problem is as, depending on how they are blocking it, google might be getting a confusing error message internally

**Reply by dpushkar24** [reply]
Correct...
So any suggestion how we can fix that?
Because receiving mail from google is very important for us.
Do we have any functionality here that if we could change our mail reviving port to 465 or 587?
Because oracle is not blocking these ports and suggesting them to use for SMTP purpose.

Please suggest

**Reply by willpower232** [reply]
You can control the port in the config file I believe or you could use iptables to listen on multiple ports simultaneously.
